### PR TITLE
Use localhost and set redis password by default

### PR DESCRIPTION
**Pull request by simon-mo**
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.

### CODE REVIEW

File: python/ray/worker.py [hunk 1]
```diff
@@ -533,15 +533,15 @@ def init(address=None,
          driver_object_store_memory=None,
          redis_max_memory=None,
          log_to_driver=True,
-         node_ip_address=None,
+         node_ip_address="127.0.0.1",
```
Comment (ericl) [inline python/ray/worker.py]: Consider making this in ray constants as well.

File: python/ray/parameter.py [hunk 2]
```diff
@@ -95,15 +95,15 @@ def __init__(self,
                  redis_shard_ports=None,
                  object_manager_port=None,
                  node_manager_port=None,
-                 node_ip_address=None,
+                 node_ip_address="127.0.0.1",
```
Comment (ericl) [inline python/ray/parameter.py]: Is this right? We only want to set it to this on local ray.init() right?

**Review by edoakes — APPROVED** [review]

**Comment by robertnishihara** [comment]
@simon-mo note that we are running Redis in unprotected mode. See

https://github.com/ray-project/ray/blob/240e8f52791d85ef6bb68cae1e1483debbbde2b2/python/ray/services.py#L898

Perhaps we should undo this?

**Comment by simon-mo** [comment]
@ericl what's your take on enabling `ray start` to set default redis password. basically `ray start` will set default password and `ray.init` will use the default password to connect to the node. no user code change required.

**Comment by simon-mo** [comment]
This simplifies test a little bit (so we don't need to explicitly set password in test_multinode and others.

**Comment by ericl** [comment]
Doesn't changing parameter.py do that already?

**Comment by simon-mo** [comment]
@ericl yeah i misunderstood we need the password for parameter.py to enable multi-node and but not the ip address

**Pull request merged**